owncloudApp/src/main/java/com/owncloud/android/ui/activity/PassCodeActivity.kt

```diff
@@ -0,0 +1,425 @@
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.owncloud.android.ui.activity
+
+import com.owncloud.android.utils.DocumentProviderUtils.Companion.notifyDocumentProviderRoots
+import android.widget.TextView
+import android.widget.EditText
+import android.os.Bundle
+import android.view.WindowManager
+import com.owncloud.android.R
+import android.widget.LinearLayout
+import android.view.View.OnFocusChangeListener
+import android.content.Intent
+import android.preference.PreferenceManager
+import android.text.TextWatcher
+import android.text.Editable
+import android.view.KeyEvent
+import android.view.View
+import android.widget.Button
+import com.owncloud.android.BuildConfig
+import com.owncloud.android.utils.PreferenceUtils
+import timber.log.Timber
+import java.lang.IllegalArgumentException
+import java.lang.StringBuilder
+import java.util.Arrays
+
+class PassCodeActivity : BaseActivity() {
+    private lateinit var mBCancel: Button
+    private lateinit var mPassCodeHdr: TextView
+    private lateinit var mPassCodeHdrExplanation: TextView
+    private lateinit var mPassCodeError: TextView
+    private val mPassCodeEditTexts = arrayOfNulls<EditText>(numberOfPassInputs)
+    private var mPassCodeDigits: Array<String?> = arrayOfNulls(numberOfPassInputs)
+    private var mConfirmingPassCode = false
+    private var mBChange = true // to control that only one blocks jump
+
+    /**
+     * Initializes the activity.
+     *
+     * @param savedInstanceState    Previously saved state - irrelevant in this case
+     */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        /// protection against screen recording
+        if (!BuildConfig.DEBUG) {
+            window.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
+        } // else, let it go, or taking screenshots & testing will not be possible
+        setContentView(R.layout.passcodelock)
+        val passcodeLockLayout = findViewById<LinearLayout>(R.id.passcodeLockLayout)
+        mBCancel = findViewById(R.id.cancel)
+        mPassCodeHdr = findViewById(R.id.header)
+        mPassCodeHdrExplanation = findViewById(R.id.explanation)
+        mPassCodeError = findViewById(R.id.error)
+
+        // Allow or disallow touches with other visible windows
+        passcodeLockLayout.filterTouchesWhenObscured =
+            PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(this)
+        mPassCodeHdrExplanation.setFilterTouchesWhenObscured(
```

Comment:
We can use the property access syntax as you did in the previous line:

```suggestion
        mPassCodeHdrExplanation.filterTouchesWhenObscured = PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(this)
```